dexlib2/src/main/java/org/jf/dexlib2/MethodHandleType.java

```diff
@@ -31,59 +31,62 @@
 
 package org.jf.dexlib2;
 
-import com.google.common.collect.Maps;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ImmutableBiMap;
 import org.jf.util.ExceptionWithContext;
 
 import javax.annotation.Nonnull;
-import java.util.Map;
 
 public class MethodHandleType {
-    public static final int STATIC_PUT = 0;
-    public static final int STATIC_GET = 1;
-    public static final int INSTANCE_PUT = 2;
-    public static final int INSTANCE_GET = 3;
-    public static final int STATIC_INVOKE = 4;
-    public static final int INSTANCE_INVOKE = 5;
+    /** A method handle that gets an instance field. */
+    public static final int REF_GET_FIELD = 3;
```

Comment:
Let's stick with the original naming scheme, which more closely matches the canonical dex specification at https://source.android.com/devices/tech/dalvik/dex-format. Unless you're planning on updating the constants there as well?

Although now that I'm looking at it, maybe switch STATIC_INVOKE, etc. to INVOKE_STATIC, to better match the constants there. Up to you though, I'm happy to make that change afterward if you prefer.

Reply:
I disagree. The naming scheme in dex is poor and confusing. Why they decided to change it from what is used in normal java I do not understand. These are much clearer. Plus with the comment above the field, there should be no confusion on what they are pointing to.

Reply:
dexlib2 is a parser for the dex format, not for java bytecode :). the dex specification is the canonical source for the dex format, and I'd prefer to stay consistent with that.

Reply:
IMO, the dex constants make more sense in the context of the dex bytecode.

e.g. there is no other reference to the "special" method nomenclature (invokespecial, etc.) in the context of a dex file. The equivalent appears to be invoke-direct. Why would we use ref_invoke_special, when the equivalent bytecode is invoke-direct?

Reply:
Also STATIC_GET = sget, INSTANCE_GET = iget, etc.

while the "GET_FIELD" and "PUT_FIELD" nomenclature appears to be referencing java's getfield and putfield instructions.

Reply:
Alright I guess. I still find the names unclear but I can put them in the exact form that dexdump uses (including the strings) which matches the ones on the dex-format page minus those missing. This is what I mean here

https://android.googlesource.com/platform/art/+/android-8.1.0_r41/dexdump/dexdump.cc#1583

Reply:
Yeah, let's keep the existing ones as-is (unless you want to change `*_INVOKE` to `INVOKE_*` -- up to you), and then the new ones should be INVOKE_DIRECT, INVOKE_CONSTRUCTOR and INVOKE_INTERFACE.